packages/swing-store/src/swingStore.js

```diff
@@ -82,6 +99,170 @@ export function makeSnapStoreIO() {
  * }} SwingStore
  */
 
+/**
+ * @typedef {[
+ *   key: string,
+ *   value: string|undefined,
+ * ]} KVPair
+ *
+ * @typedef {object} SwingStoreExporter
+ *
+ * Allows export of data from a swingStore as a fixed view onto the content as
+ * of the most recent commit point at the time the exporter was created.  The
+ * exporter may be used while another SwingStore instance is active for the same
+ * DB, possibly in another thread or process.  It guarantees that regardless of
+ * the concurrent activity of other swingStore instances, the data representing
+ * the commit point will stay consistent and available.
+ *
+ * @property {() => Iterable<KVPair>} getKVData
+ *
+ * Get a full dump of KV data from the swingStore. This represents both the
+ * KVStore (excluding host and local prefixes), as well as any data needed to
+ * validate all artifacts, both current and historical. As such it represents
+ * the root of trust for the application.
+ *
+ * Content of validation data (with supporting entries for indexing):
+ * - export.snapshot.${vatID}.${endPos} = ${{ vatID, endPos, hash, size });
+ * - export.snapshot.${vatID}.current = `snapshot.${vatID}.${endPos}`
```

Comment:
Why use the artifact key instead of the metadata ?
```suggestion
 * - export.snapshot.${vatID}.current = ${{ vatID, endPos, hash, size }}
```

Reply:
I agree that this wants to be more metadata-shaped than name-of-an-artifact -shaped. It does need to be a single string, though, and safely parseable back into the metadata with which the new DB row can be populated.. I'm not sure if the proposed `${{ ..` line would do that.

Reply:
Because it makes one fewer line of code that needs to know about the format/content of the metadata.